src/Libraries/DSCPython/CPythonEvaluator.cs

```diff
@@ -43,39 +43,55 @@ public static class CPythonEvaluator
             if (code != prev_code)
             {
                 Python.Included.Installer.SetupPython().Wait();
-                PythonEngine.Initialize();
-                using (Py.GIL())
+
+                if (!PythonEngine.IsInitialized)
                 {
-                    using (PyScope scope = Py.CreateScope())
-                    {
-                        int amt = Math.Min(bindingNames.Count, bindingValues.Count);
+                    PythonEngine.Initialize();
+                    PythonEngine.BeginAllowThreads();
```

Comment:
@reddyashish - you will need to tell me what is happening here 😉 - did one of your new tests introduce a deadlock? 
@Dewb FYI

Reply:
Just saw your comment, in what situations (which test jobs) are multiple test fixtures creating multiple python engines in the same process?

Reply:
I think the explanation is here https://github.com/pythonnet/pythonnet/wiki/Threading

> The Python interpreter only actually runs single-threaded. You can create additional threads, but only one thread at a time has the Global Interpreter Lock (GIL), and only that thread can run. 

So that would mean only one thread should call PythonNet at a time. Not sure what test was failing though.

Reply:
Sorry guys, was waiting for the self-serve. Tested it locally. I wrote the explanation before.

Reply:
It isn't happening with a single test but when the PythonEngine is initialized from 2 different test fixtures.